src/components/shared/FormDialog.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from 'react';
+
+// material ui
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Dialog from '@material-ui/core/Dialog';
+import CheckIcon from '@material-ui/icons/Check';
+import ClearIcon from '@material-ui/icons/Clear';
+import { green } from '@material-ui/core/colors';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import Input from '@material-ui/core/Input';
+import InputLabel from '@material-ui/core/InputLabel';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import IconButton from '@material-ui/core/IconButton';
+import Visibility from '@material-ui/icons/Visibility';
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
+import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
+
+import {EmailForm} from './EmailForm';
+import {PasswordForm} from './PasswordForm';
+import {FormField} from './FormField';
+import {EmailState, PasswordState} from '../../shared/types/models';
+import {isValidEmail} from '../../shared/helper_functions';
+
+export interface AddUserProps {
+    title: string;
+    isOpen: boolean;
+    handleClose: (value: boolean, firstname: string, lastname: string, email: string, password: string) => void;
+}
+
+export function FormDialog({ handleClose, isOpen, title }: AddUserProps) {
+  const classes = useStyles();
+  const [firstName, setFirstName] = useState<string>('');
+  const [lastName, setLastName] = useState<string>('');
+  const [email, setEmail] = useState<EmailState>(
+    { value: '', valid: true, message: null },
+  );
+  const [password, setPassword] = useState<PasswordState>({
+    value: '',
+    valid: true,
+    visible: false,
+    message: null,
```

Comment:
@hchiam its actually not erroring haha

Reply:
@hchiam that's how shaun has it in his login page so that what i did lol and it seems to work

Reply:
@hchiam its his component

Reply:
Yeah null is actually allowed for every type in typescript. It's just one of those things. The reason I had message be null is because to me, an error should be not exist until there is one. An empty string usually signifies that the value is known, just empty. A null means there is no value. To me there is no error, thus no error message. Thus null. Also it's easier to write with